lmnet/lmnet/datasets/bdd100k.py

```diff
@@ -1,16 +1,24 @@
 # -*- coding: utf-8 -*-
-import functools
-
-from os.path import join, splitext, basename
+import math
+import os
+import pymp
+import time
```

Comment:
pymp is not a dependence module of blueoil/lmnet

Reply:
> You can suggest a better way if you have something in mind.

You can use `multiprocessing` module directly, which is a standard library, well tested and don't require extra dependency.

BTW, I measured the performance of parsing annotation files on our 22th server, once json files are on the disk cache, I got the following result:

```
elapsed time: 16.161115646362305  for  70000  images
elapsed time: 2.5325820446014404  for  10000  images
```
Compared to your 1206sec, these numbers are very small 🤔Could you run the experiment twice, to check the effectiveness of disk cache?

Reply:
@tkng San: 
I talked with @iizukak san and I'll remove the `pymp` package dependency. And you're right once the json is in the cache it will be significantly faster but it also depends on the system. I've tested in server 5, which may be slower than `srv22`

Reply:
@tkng:
I've run the training on `lm-srv-05` with the following results:
* 1st run : 
  * 70000 images : 997.87 sec
  * 10000 images : 24.51 sec
* 2nd run :
  * 70000 images : 1001.92 sec  
  * 10000 images : 25.61 sec
* 3rd run [from pickle files] :
  * 70000 images : < 1 sec
  * 10000 images : < 1 sec

Reply:
Could you tell me how to reproduce the result? I cannot understand why the result of the 1st and 2nd trial are almost same... 2nd trial should be much faster, as we expect the speed up gain by disk cache...

In my case, you can reproduce by the following steps:
1. login to lmsvr05
2. cd to /home/tokunaga/blueoil/lmnet
3. `PYTHONPATH=. CUDA_VISIBLE_DEVICES="3" python3.6 executor/train.py -c configs/core/object_detection/lm_fyolo_bdd100k.py -i bdd100k_test`

Reply:
@tkng San:
:open_mouth: I'm confused.
Server 5 has 2 GPUs, How can you run with `CUDA_VISIBLE_DEVICES="3"` without an error.

Anyway, Please check the pickle file at `blueoil/lmnet/dataset/bdd100k_{train/val}.pickle`. Please remove them.

Reply:
@tkng San:

> NOTE : I appologize for long discussion.

I've tried to reproduce your case with the current master branch of blueoil on server 5. Here is my result.
* The training seems to be running but the dataset clearly not populated with data. As in the error, for both training and validation dataset_iterator gives the following error:
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/hasibur/blueoil/lmnet/lmnet/datasets/dataset_iterator.py", line 163, in run
    self.loop_body()
  File "/home/hasibur/blueoil/lmnet/lmnet/datasets/dataset_iterator.py", line 139, in loop_body
    task_list = self.gen_task(self.dataset.batch_size * 8)
  File "/home/hasibur/blueoil/lmnet/lmnet/datasets/dataset_iterator.py", line 123, in gen_task
    data_id = self.data_ids.pop()
IndexError: pop from empty list

Exception in thread Thread-5:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/hasibur/blueoil/lmnet/lmnet/datasets/dataset_iterator.py", line 163, in run
    self.loop_body()
  File "/home/hasibur/blueoil/lmnet/lmnet/datasets/dataset_iterator.py", line 139, in loop_body
    task_list = self.gen_task(self.dataset.batch_size * 8)
  File "/home/hasibur/blueoil/lmnet/lmnet/datasets/dataset_iterator.py", line 123, in gen_task
    data_id = self.data_ids.pop()
IndexError: pop from empty list
```
* The reason of the error is in the `blueoil/lmnet/dataset/bdd100k.py` dataset file the `_init_files_and_annotations` function expect the annotation in json file for each image with the same filename but if you download the data from BDD website and extract, instead of getting individual json for each image you'll get 1.5GB json files, one for training and one for validation. So, the `bdd100k.py` dataset file in current blueoil master branch no longer works.

But we already identified the bottleneck and improved performance. Please checkout the current update of my pull request, now it takes less than 1 min. The biggest improvement we've got is as follows:
* **Changing `List` -> `Dict`** : From Python 3.5 dict is significantly improved in performance
* **Changing `List` -> `Set`** : In case where order is not required set gives more performance by using hash matching.